samcli/lib/config/file_manager.py

```diff
@@ -260,3 +260,74 @@ def _to_yaml(document: dict) -> Any:
         with StringIO() as stream:
             YamlFileManager.yaml.dump(document, stream)
             return YamlFileManager.yaml.load(stream.getvalue())
+
+
+class JsonFileManager(FileManager):
+    """
+    Static class to read and write json files.
+    """
+
+    file_format = "JSON"
+
+    @staticmethod
+    def read(filepath: Path) -> Any:
+        """
+        Read a JSON file at a given path.
+
+        Parameters
+        ----------
+        filepath: Path
+            The Path object that points to the file to be read.
+
+        Returns
+        -------
+        Any
+            A dictionary representation of the contents of the JSON document.
+        """
+        json_file = {}
+        try:
+            json_file = json.loads(filepath.read_text())
+        except OSError as e:
+            LOG.debug(f"OSError occurred while reading {JsonFileManager.file_format} file: {str(e)}")
+        except json.JSONDecodeError as e:
+            raise FileParseException(e) from e
+        return json_file
+
+    @staticmethod
+    def write(document: dict, filepath: Path):
+        """
+        Write a dictionary or dictionary-like object to a JSON file.
+
+        Parameters
+        ----------
+        document: dict
+            The JSON object to write.
+        filepath: Path
+            The final location for the document to be written.
+        """
+        if not document:
+            LOG.debug("No document given to JsonFileManager to write.")
+            return
+
+        with filepath.open("w") as file:
+            json.dump(document, file)
```

Comment:
Could we also specify `indent=2` to the dump method so that the file will be more easily editable?

Reply:
Now that you mention it, would it be better to specify it as a local parameter for `JsonFileManager`? I'm thinking that way we can tweak indentation values if need be, and prevent magic numbers.

Reply:
That sounds like a good improvement to have it in a variable